core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -169,7 +160,7 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def instantiateClass(cls: Class[_], outer: AnyRef, inInterpreter: Boolean): AnyRef = {
     // logInfo("Creating a " + cls + " with outer = " + outer)
-    if (!inInterpreter) {
+    if (inInterpreter) {
```

Comment:
Not exactly.
This is why a put a note in the description asking for extra attention.

Previous behavior in case of a `ClassNotFoundException` was to follow the same path as when the presence of the interpreter was successfully confirmed. This looked not logical to me and I decided to correct while it being at it.

Now, when the exception is raised in evaluation of `isInInterpreter`, the code in `org.apache.spark.util.ClosureCleaner#instantiateClass` will follow the same path as when `interpClass.getMethod("interp").invoke(null)` returned `null` indicating presence of the REPL wasn't detected.

Hope, this clarifies my intention.